sql/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -190,7 +187,7 @@ public Settings analyzerSettings() {
 
     public void validate() {
 
-        if ("not_analyzed".equals(analyzer) && !STRING_TYPES.contains(dataType)) {
+        if (analyzer != null && !"not_analyzed".equals(analyzer) && !dataType.equals("string")) {
```

Comment:
I think I added the STRING_TYPES because of https://github.com/crate/crate/blob/five/sql/src/main/java/io/crate/analyze/TableElementsAnalyzer.java#L245